File: hello-typescript/src/client/util/VerificationModule.ts
/**
 * Verification module for comparing original and downloaded files.
 * Performs file size and checksum verification.
 * Matches Java VerificationModule functionality.
 */

import { FileManager } from "../core/FileManager";

export class VerificationReport {
  private passed: boolean;
  private originalSize: number;
  private downloadedSize: number;
  private originalChecksum: string;
  private downloadedChecksum: string;

  constructor(
    passed: boolean,
    originalSize: number,
    downloadedSize: number,
    originalChecksum: string,
    downloadedChecksum: string,
  ) {
    this.passed = passed;
    this.originalSize = originalSize;
    this.downloadedSize = downloadedSize;
    this.originalChecksum = originalChecksum;
    this.downloadedChecksum = downloadedChecksum;
  }

  isVerificationPassed(): boolean {
    return this.passed;
  }

  printReport(): void {
    console.log("\n=== Verification Report ===");
    console.log(`Original size: ${this.originalSize} bytes`);
    console.log(`Downloaded size: ${this.downloadedSize} bytes`);
    console.log(`Original checksum (SHA-256): ${this.originalChecksum}`);
    console.log(`Downloaded checksum (SHA-256): ${this.downloadedChecksum}`);

    if (this.passed) {
      console.log("✓ File verification PASSED - Files are identical");
    } else {
      console.log("✗ File verification FAILED");
      if (this.originalSize !== this.downloadedSize) {
        console.log(`  Reason: File size mismatch`);
      }
      if (this.originalChecksum !== this.downloadedChecksum) {
        console.log("  Reason: Checksum mismatch");
      }
    }
    console.log("===========================\n");
  }
}

export class VerificationModule {
  private fileManager: FileManager;

  constructor() {
    this.fileManager = new FileManager();
  }

  async generateReport(
    originalPath: string,
    downloadedPath: string,
  ): Promise<VerificationReport> {
    // Get file sizes
    const originalSize = await this.fileManager.getFileSize(originalPath);
    const downloadedSize = await this.fileManager.getFileSize(downloadedPath);

    // Calculate checksums
    const originalChecksum =
      await this.fileManager.calculateChecksum(originalPath);
    const downloadedChecksum =
      await this.fileManager.calculateChecksum(downloadedPath);

    // Compare
    const passed =
      originalSize === downloadedSize &&
      originalChecksum === downloadedChecksum;

    return new VerificationReport(
      passed,
      originalSize,
      downloadedSize,
      originalChecksum,
      downloadedChecksum,
    );
  }
}

File: hello-python/src/client/util/verification_module.py
"""File verification module for comparing original and downloaded files"""

from typing import Dict, Any
from .. import logger
from ..core import file_manager


async def verify(original_path: str, downloaded_path: str) -> Dict[str, Any]:
    """Verify file integrity"""
    logger.info(f"Original file: {original_path}")
    logger.info(f"Downloaded file: {downloaded_path}")
    
    # Get file sizes
    original_size = await file_manager.get_file_size(original_path)
    downloaded_size = await file_manager.get_file_size(downloaded_path)
    
    logger.info(f"Original size: {original_size} bytes")
    logger.info(f"Downloaded size: {downloaded_size} bytes")
    
    # Calculate checksums
    original_checksum = await file_manager.calculate_checksum(original_path)
    downloaded_checksum = await file_manager.calculate_checksum(downloaded_path)
    
    logger.info(f"Original checksum (SHA-256): {original_checksum}")
    logger.info(f"Downloaded checksum (SHA-256): {downloaded_checksum}")
    
    # Compare
    passed = original_size == downloaded_size and original_checksum == downloaded_checksum
    
    return {
        'passed': passed,
        'original_size': original_size,
        'downloaded_size': downloaded_size,
        'original_checksum': original_checksum,
        'downloaded_checksum': downloaded_checksum
    }


class VerificationModule:
    """Module for file verification operations"""
    
    @staticmethod
    async def verify_files(original_path: str, downloaded_path: str) -> Dict[str, Any]:
        """Verify file integrity"""
        return await verify(original_path, downloaded_path)
